src/setup/rgw.py

```diff
@@ -137,22 +137,8 @@ def config(conf: MappedConf):
         # config phase since data pod starts before server pod.
         # Try HAX endpoint from data pod of same node first & if it doesnt work,
         # from other data pods in cluster
-        current_data_node = socket.gethostname().replace('server', 'data')
```

Comment:
why this move is needed exactly to line 530? Please add explanation here (not in code)

Reply:
It is used both in 140 & 530. Hence created a new function..
when recursively called data_pod_hostname changes to other pod's hostname

Reply:
Code is optimised by creating two separate functions